modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormInstanceSettings.java

```diff
@@ -139,6 +139,15 @@
 	)
 	public String emailToAddress();
 
+	@DDMFormField(
+		label = "%expiration-date", type = "date",
+		validationErrorMessage = "%please-enter-a-valid-expiration-date-only-future-dates-are-accepted",
+		validationExpression = "futureDates(expirationDate, \"{parameter}\")",
+		validationExpressionName = "futureDates",
```

Comment:
@rodrigopaulino, can you please give more details about the purpose of the parameter `validationExpressionName`?

Thanks!

Reply:
Yeah, of course!

This is necessary for the same reason DDMValidations have [names](https://github.com/liferay/liferay-portal/blob/c5a2f60bc87ff963fe1b71b63a0519e1ae913cad/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/validation/DDMValidation.java#L26) and so DDMFormFieldValidationExpression can replace the parameter accordingly ([here](https://github.com/liferay/liferay-portal/blob/c5a2f60bc87ff963fe1b71b63a0519e1ae913cad/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormFieldValidationExpression.java#L76)).

Hope this clarifies.

Reply:
Thanks for your clarification, pal! :wink: